// Generated by GitHub - foxglove/foxglove-sdk: Libraries and schemas for Foxglove

syntax = "proto3";

import "Pose.proto";
import "google/protobuf/timestamp.proto";
import "header.proto";
import "Vector3.proto";

package foxglove;

// A timestamped pose for an object or reference frame in 3D space
message PoseInFrame {
  // Timestamp of pose
  google.protobuf.Timestamp timestamp = 1;

  // Frame of reference for pose position and orientation
  string frame_id = 2;

  // Pose in 3D space
  foxglove.Pose pose = 3;
  // linear vel
  foxglove.Vector3 linear_vel= 4;
  // angular_vel
  foxglove.Vector3 angular_vel = 5;

  arnold.common.proto.Header header = 6;
}